objectivec/GPBMessage_PackagePrivate.h

```diff
@@ -42,6 +44,125 @@
 
 #import "GPBBootstrap.h"
 
+#if defined(__LP64__) && __LP64__
+#define GPB_ASM_PTR " .quad "
+#define GPB_ASM_PTRSIZE " 8 "
+#define GPB_ASM_ONLY_LP64(x) x
+#define GPB_ALIGN_SIZE " 3 "
+#define GPB_DESCRIPTOR_METHOD_TYPE "@16@0:8"
+#else  // __LP64__
+#define GPB_ASM_PTR " .long "
+#define GPB_ASM_PTRSIZE " 4 "
+#define GPB_ASM_ONLY_LP64(x)
+#define GPB_ALIGN_SIZE " 2 "
+#define GPB_DESCRIPTOR_METHOD_TYPE "@8@0:4"
+#endif  // __LP64__
+
+#define GPB_MESSAGE_CLASS_NAME_RAW GPBMessage
+#define GPB_MESSAGE_CLASS_NAME GPBStringifySymbol(GPB_MESSAGE_CLASS_NAME_RAW)
+
+#if !__has_feature(objc_arc)
+#define GPB_CLASS_FLAGS " 0x00 "
+#define GPB_METACLASS_FLAGS " 0x01 "
+#else
+// 0x80 denotes that the class supports ARC.
+#define GPB_CLASS_FLAGS " 0x80 "
+#define GPB_METACLASS_FLAGS " 0x81 "
+#endif
+
+// This generates code equivalent to:
+// ```
+// @implementation _name_
+// + (GPBMessage*)descriptor {
```

Comment:
`+descriptor` returns `GPBDescriptor*`, not `GPBMessage*`

Reply:
Fixed. Thanks